qiskit/ignis/verification/randomized_benchmarking/rb_utils.py

```diff
@@ -445,3 +471,163 @@ def calculate_2q_epg(gate_per_cliff: Dict[int, Dict[str, float]],
 
     raise QiskitError('Two qubit gate %s is not included in the `gate_per_cliff`. '
                       'Set correct `two_qubit_name` or use 2Q RB gate count.' % two_qubit_name)
+
+
+def calculate_1q_epc(gate_per_cliff: Dict[int, Dict[str, float]],
+                     epgs: Dict[str, float],
+                     qubit: int) -> float:
+    r"""
+    Convert error per gate (EPG) into error per Clifford (EPC) of single qubit basis gates.
+
+    Given that we know the number of gates per Clifford :math:`N_i` and those EPGs,
+    we can predict EPC of that RB sequence:
+
+    .. math::
+
+        EPC = 1 - \prod_i \left( 1 - EPG_i \right)^{N_i}
+
+    To run this function, you need to know EPG of every single qubit basis gates.
+    For example, when you prepare 1Q RB experiment with appropriate error model,
+    you can define EPG of those basis gate set. Then you can estimate the EPC of
+    prepared RB sequence without running experiment.
+
+    .. jupyter-execute::
+
+        import qiskit.ignis.verification.randomized_benchmarking as rb
+
+        # gate counts of your 1Q RB experiment
+        gpc = {0: {'cx': 0, 'u1': 0.13, 'u2': 0.31, 'u3': 0.51}}
+
+        # EPGs from error model
+        epgs_q0 = {'u1': 0, 'u2': 0.001, 'u3': 0.002}
+
+        # calculate 1Q EPC
+        epc = rb.rb_utils.calculate_1q_epc(
+            gate_per_cliff=gpc,
+            epgs=epgs_q0,
+            qubit=0)
+
+        print(epc)
+
+    Args:
+        gate_per_cliff: dictionary of gate per Clifford. see :func:`gates_per_clifford`.
+        epgs: EPG of single qubit gates estimated by error model.
+        qubit: index of qubit to calculate EPC.
+
+    Returns:
+        EPG of 2Q gate.
+
+    Raises:
+        QiskitError: when specified ``qubit`` is not included in the gate count dictionary
+
+    """
+    if qubit not in gate_per_cliff:
+        raise QiskitError('Qubit %d is not included in the `gate_per_cliff`' % qubit)
+
+    fid = 1
+    gpc_per_qubit = gate_per_cliff[qubit]
+
+    for gate_name, epg in epgs.items():
+        n_gate = gpc_per_qubit.get(gate_name, 0)
+        fid *= (1 - epg) ** n_gate
+
+    return 1 - fid
+
+
+def calculate_2q_epc(gate_per_cliff: Dict[int, Dict[str, float]],
+                     epg_2q: float,
+                     qubit_pair: List[int],
+                     list_epgs_1q: List[Dict[str, float]],
```

Comment:
Can this be optional like the `calculage_2q_epg` function?

Reply:
I think it doesn't make sense. for `calculate_2q_epg` it returns epg = `epc_2q/n_gate_2q` when `list_epgs_1q` is None. for this, it returns epc = `3/4*(1-(1-4/3*epg_2q)**n_gate_2q)`. when `epg_2q` << 1 it can be approximated to epc = `epg_2q*n_gate_2q`.